### ISSUE TITLE
Sepolia SIGILL: illegal instruction

### ISSUE THREAD

**Issue by vivienbcr** [prysmaticlabs/prysm#13567]
### Describe the bug

I'm trying to run latest version of beacon-chain (`beacon-chain version Prysm/v4.2.1/59b310a2216c57fcf67ea0fdec739dad07aeec8b`)  on sepolia network and get error `illegal instruction`.

According to the changelog, this error seems expected. Do you plan to support older generations of processors or release some sort of portable build?

```
Feb 01 07:22:14 beacon-chain[2582544]: time="2024-02-01 07:22:14" level=info msg="Peer summary" activePeers=8 inbound=5 outbound=3 prefix=p2p
Feb 01 07:23:14 beacon-chain[2582544]: time="2024-02-01 07:23:14" level=info msg="Peer summary" activePeers=7 inbound=7 outbound=0 prefix=p2p
Feb 01 07:23:27 beacon-chain[2582544]: time="2024-02-01 07:23:27" level=info msg="Processing block batch of size 60 starting from  0xd4e09fd3... 4243456/4253217 - estimated time remaining 54m13s" blocksPerSecond=3.0 peers=7 prefix=initial-sync
Feb 01 07:23:34 beacon-chain[2582544]: SIGILL: illegal instruction
Feb 01 07:23:34 beacon-chain[2582544]: PC=0x21680e1 m=8 sigcode=2
Feb 01 07:23:34 beacon-chain[2582544]: instruction bytes: 0xc4 0x43 0x7b 0xf0 0xe9 0xe 0xc4 0xe3 0x61 0xf 0xe2 0x4 0xc4 0x63 0x7b 0xf0
```

Cpu info :

```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   46 bits physical, 48 bits virtual
CPU(s):                          24
On-line CPU(s) list:             0-23
Thread(s) per core:              1
Core(s) per socket:              12
Socket(s):                       2
NUMA node(s):                    2
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           62
Model name:                      Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
Stepping:                        4
CPU MHz:                         1403.500
CPU max MHz:                     3500.0000
CPU min MHz:                     1200.0000
BogoMIPS:                        5400.06
L1d cache:                       768 KiB
L1i cache:                       768 KiB
L2 cache:                        6 MiB
L3 cache:                        60 MiB
NUMA node0 CPU(s):               0-11
NUMA node1 CPU(s):               12-23
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon peb
                                 s bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic popcnt tsc_deadline_
                                 timer aes xsave avx f16c rdrand lahf_lm cpuid_fault epb pti intel_ppin ssbd ibrs ibpb stibp fsgsbase smep erms xsaveopt dtherm ida arat pln pts md_clear flush_l1d
```

### Has this worked before in a previous version?

```text
Work well with 4.2.0
```

### 🔬 Minimal Reproduction

1. Get release 4.2.1 from github
or 
2. clone, checkout 4.2.1, bazel build cmd/beacon-chain --config=release

### Error

[https://pastecode.io/s/0u4jbvrt](https://pastecode.io/s/0u4jbvrt)

### Platform(s)

Linux (x86)

### What version of Prysm are you running? (Which release)

beacon-chain version Prysm/v4.2.1/59b310a2216c57fcf67ea0fdec739dad07aeec8b

**Comment by Jony321** [comment]
I have a similar problem.
I asked to re-open my problem (https://github.com/prysmaticlabs/prysm/issues/13485), updating to 4.2.1 did not solve the problem.

**Comment by prestonvanloon** [comment]
https://github.com/prysmaticlabs/prysm/pull/13569 introduces the fix from prysmaticlabs/gohashtree#16

**Comment by prestonvanloon** [comment]
Fixed in v4.2.2-rc.0
